include/rmm/device_buffer.hpp

```diff
@@ -315,11 +315,19 @@ class device_buffer {
   void* data() noexcept { return _data; }
 
   /**
-   * @brief Returns size in bytes that was requested for the device memory
-   * allocation
+   * @brief Returns size in bytes of the underlying device memory storage.
```

Comment:
The somewhat awkward original documentation was my attempt to differentiate `size()` and `capacity()`. With this change, I feel that nuance is lost.

Reply:
OK, I updated the docs to be based more on the size() and capacity() docs of `std::vector`, which is exactly what you had done in `rmm::device_uvector` already.